relay-quotas/src/global.rs

```diff
@@ -48,40 +70,67 @@ impl GlobalRateLimits {
 struct Key {
     prefix: String,
     window: u64,
+    namespace: Option<MetricNamespace>,
```

Comment:
is it worth the extra string allocation though? We only use `Key` when we put a new key into the hashmap, otherwise we use keyref which is a cheap copy type. If we change the 'Key' here we'd have to do away with the cheap keyref type.